api/grpcserver/grpc_server.go

```diff
@@ -77,12 +82,51 @@ func (s *GrpcServer) GetAllUsersLikeUsername(_ context.Context, username *proto.
 }
 
 //GetUserWithUsername returns one User that equals the given username
-func (s *GrpcServer) GetUserWithUsername(_ context.Context, username *proto.UserSearchRequest) (*proto.User, error) {
+func (s *GrpcServer) GetUserWithUsername(_ context.Context, username *proto.UserNameRequest) (*proto.User, error) {
 	u := &proto.User{}
 	fmt.Println(username)
 
-	row := s.db.QueryRow("SELECT user_name, real_name, bio, avatar_url FROM users WHERE user_name = $1", username.UserName)
+	err := s.db.QueryRow("SELECT id, user_name, real_name, bio, avatar_url FROM users WHERE user_name = $1", username.UserName).Scan(&u.Id, &u.UserName, &u.RealName, &u.Bio, &u.AvatarUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := s.db.Query("Select follows.to_id as id, users.user_name from follows join users on follows.to_id=users.id where follows.from_id=$1", u.Id)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		followingUser := proto.User{}
+
+		rows.Scan(&followingUser.Id, &followingUser.UserName)
+
+		u.Followings = append(u.Followings, &followingUser)
+
+	}
 
-	row.Scan(&u.UserName, &u.RealName, &u.Bio, &u.AvatarUrl)
 	return u, nil
 }
+
+//GetInstaPostsWithUserId returns all Instagram Posts of a User
+func (s *GrpcServer) GetInstaPostsWithUserId(_ context.Context, request *proto.UserIdRequest) (*proto.InstaPostsResponse, error) {
+	res := &proto.InstaPostsResponse{}
+
+	rows, err := s.db.Query("SELECT id, post_id, short_code, picture_url FROM posts WHERE user_id=$1", request.UserId)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	res.UserId = request.UserId
+	for rows.Next() {
+		post := proto.InstaPost{}
+
+		rows.Scan(&post.Id, &post.PostId, &post.ShortCode, &post.ImgUrl)
```

Comment:
`picture_url` is not endlessly valid; the signature in the url is only valid for a few days/weeks.    
Once #67 is merged, what we should do here is generate a presigned url to the image we have downloaded to S3 and return that as the `ImgUrl` out: https://docs.min.io/docs/golang-client-api-reference#PresignedGetObject
:)